Greg was 14, Marcia was 12, Peter was 11, Jan was 10, Bobby was 8, and Cindy was 6 when they started playing the Brady kids on The Brady Bunch. Cousin Oliver was 8 years old when he joined the show. What are the mean, median, and mode of the kids' ages when they first appeared on the show? What are the variance, standard deviation, and standard error?

In [19]:
import numpy as np
import pandas as pd
import matplotlib


In [64]:
d = {'Name': ['Greg', 'Marcia', 'Peter', 'Jan', 'Bobby', 'Cindy', 'Oliver'], 
     'Age' : [14,12,11,10,8,6,8]}

In [65]:
data = pd.DataFrame(d)

In [56]:
data

Name  Age
0    Greg   14
1  Marcia   12
2   Peter   11
3     Jan   10
4   Bobby    8
5   Cindy    6
6  Oliver    8

In [23]:
data.mean()

Age    9.857143
dtype: float64

In [24]:
data.median()

Age    10.0
dtype: float64

In [25]:
data.Age.mode()

0    8
dtype: int64

In [26]:
data.std(ddof=1)

Age    2.734262
dtype: float64

In [27]:
data.sem(ddof=1)

Age    1.033454
dtype: float64

In [30]:
np.var(data.Age)

6.408163265306122

In [31]:
data.describe()

Age
count   7.000000
mean    9.857143
std     2.734262
min     6.000000
25%     8.000000
50%    10.000000
75%    11.500000
max    14.000000

Using these estimates, if you had to choose only one estimate of central tendency and one estimate of variance to describe the data, which would you pick and why?

Answer: I would say that mean is the best estimate of central tendency because it is very close to the median, which means that it is pretty accurate and has a low variance.
For the estimate of variance, I would choose the standard deviation because it shows how much the ages are different. 

Next, Cindy has a birthday. Update your estimates- what changed, and what didn't?

In [66]:
newdata = data.replace({'Name': 6}, 7)

In [67]:
newdata

Name  Age
0    Greg   14
1  Marcia   12
2   Peter   11
3     Jan   10
4   Bobby    8
5   Cindy    7
6  Oliver    8

In [71]:
newdata.describe()

Age
count   7.000000
mean   10.000000
std     2.516611
min     7.000000
25%     8.000000
50%    10.000000
75%    11.500000
max    14.000000

In [72]:
data.describe()

Age
count   7.000000
mean    9.857143
std     2.734262
min     6.000000
25%     8.000000
50%    10.000000
75%    11.500000
max    14.000000

The mean went up and the std did as well. Same with the min age. The 25% 50% 75% and Max all stayed the same though. 

In [74]:
newerdata = newdata.drop(6)

In [79]:
d2 = pd.DataFrame({'Name': ['Jessica'], 'Age' : [1]})
newerdata = newerdata.append(d2)

In [83]:
newerdata.describe()

Age
count   7.000000
mean    9.000000
std     4.242641
min     1.000000
25%     7.500000
50%    10.000000
75%    11.500000
max    14.000000

The standard deviation has drastically jumpped up. I would probably use median as the central tendancy now.

In [85]:
(20+23+17+5)/4

16.25

In [87]:
(20+17)/2

18.5

I think about 19% of adult Americans watched the Brady Bunch. I feel like the SciPhi fans are a smaller amount of peolple than the TV guide magizine and entertainment weekly. So I didn't want to do mean because the SciFi 5% dropped it. So I used median. 